### PR TITLE
Updated Coevolution Lecture

### PR DESCRIPTION
**Pull request by AzizAbdi**
I thought I pushed the changes earlier. My bad! I also had to create a new branch because I can't seem to access my earlier branch. Hope that's not an issue. Please let me know if I need to make any further changes. Thanks!

:new: :bomb: :mortar_board: :memo: :robot: :lipstick: :bug: :telescope: :art: Add relevant emojis to title. [See style guide.](https://github.com/jameshughes89/cs101/blob/main/CONTRIBUTING.md#style-guidelines).

### Related Issues or PRs

https://github.com/jameshughes89/cs4XX-EvolutionaryComputation/pull/257

### What

Updated student lecture on Coevolution

### Why

Prof recommendations

### Where

student-lecture/coevolution

### How

Made the required changes, and added made few other changes to the lecture.

### Testing

NA

### Additional Notes

None

### CODE REVIEW

File: site/student-lectures/Coevolution/Coevolution.rst [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+#. Many birds avoid certain butterflies that taste bad, like the Monarch, which acquires its bitter taste from feeding on milkweed. 
+#. To deter birds, some butterflies have evolved to resemble the Monarch.
+#. These birds then learn (evolve) to distinguish Monarch bbutterflies from other kinds of butterflies.
+#. Leading to an ongoing coevolutionary cycle.
+
+.. figure:: coevolution_butterflies.png
+   :width: 400 px
+   :align: center
+
+   An illustration of the coevolutionary dynamics between predators (birds) and prey (butterflies), showcasing the evolutionary adaptations in response to each other. From: https://www.expii.com/t/coevolution-definition-examples-10963
```
Comment (jameshughes89) [inline site/student-lectures/Coevolution/Coevolution.rst:52]: Remove the URL from the caption and add a target to the figure that sends people to the url.

**Review by jameshughes89 — CHANGES_REQUESTED** [review]
1. It is unclear if you added another example with well defined steps, as requested in the review of your other PR (#257)
2. You said "Testing" was "N/A". You absolutely must test your code before you submit a PR for review --- build the site and see if it renders properly/breaks the website. 
3.

**Comment by AzizAbdi** [comment]
For the example one, where should I add a new example? In the 'Biological Systems' section or the 'CEA Models' section?

**Comment by jameshughes89** [comment]
> For the example one, where should I add a new example? In the 'Biological Systems' section or the 'CEA Models' section?

That's up to you.

**Comment by AzizAbdi** [comment]
When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!

**Comment by jameshughes89** [comment]
> When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!

You can (a) build it locally or download the artifact from the action.

**Comment by AzizAbdi** [comment]
> > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> 
> You can (a) build it locally or download the artifact from the action.

I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?

**Comment by jameshughes89** [comment]
> > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > 
> > 
> > You can (a) build it locally or download the artifact from the action.
> 
> I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?

Just open the `index.html` file and navigate the website there. It will be a local version.

**Comment by AzizAbdi** [comment]
> > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > 
> > > 
> > > You can (a) build it locally or download the artifact from the action.
> > 
> > 
> > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> 
> Just open the `index.html` file and navigate the website there. It will be a local version.

Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.

**Comment by jameshughes89** [comment]
> > > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > > 
> > > > 
> > > > You can (a) build it locally or download the artifact from the action.
> > > 
> > > 
> > > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> > 
> > 
> > Just open the `index.html` file and navigate the website there. It will be a local version.
> 
> Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.

First, you will need to add your lecture to the index in your PR
Second, did you extract the website before opening the index? If not, that could be the issue.

**Comment by AzizAbdi** [comment]
> > > > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > > > 
> > > > > 
> > > > > You can (a) build it locally or download the artifact from the action.
> > > > 
> > > > 
> > > > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> > > 
> > > 
> > > Just open the `index.html` file and navigate the website there. It will be a local version.
> > 
> > 
> > Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.
> 
> First, you will need to add your lecture to the index in your PR Second, did you extract the website before opening the index? If not, that could be the issue.

How do add my lecture to the PR index? And yes you're right, the error message was because of not extracting the files. Thanks!

**Comment by jameshughes89** [comment]
> > > > > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > > > > 
> > > > > > 
> > > > > > You can (a) build it locally or download the artifact from the action.
> > > > > 
> > > > > 
> > > > > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> > > > 
> > > > 
> > > > Just open the `index.html` file and navigate the website there. It will be a local version.
> > > 
> > > 
> > > Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.
> > 
> > 
> > First, you will need to add your lecture to the index in your PR Second, did you extract the website before opening the index? If not, that could be the issue.
> 
> How do add my lecture to the PR index? And yes you're right, the error message was because of not extracting the files. Thanks!

Just look at the index currently and see how the others did it.

**Comment by AzizAbdi** [comment]
> > > > > > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > > > > > 
> > > > > > > 
> > > > > > > You can (a) build it locally or download the artifact from the action.
> > > > > > 
> > > > > > 
> > > > > > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> > > > > 
> > > > > 
> > > > > Just open the `index.html` file and navigate the website there. It will be a local version.
> > > > 
> > > > 
> > > > Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.
> > > 
> > > 
> > > First, you will need to add your lecture to the index in your PR Second, did you extract the website before opening the index? If not, that could be the issue.
> > 
> > 
> > How do add my lecture to the PR index? And yes you're right, the error message was because of not extracting the files. Thanks!
> 
> Just look at the index currently and see how the others did it.

I have build and tested the website on my system, and it's working well with my lecture included in it. But I don't have see any changes to push because I haven't added any new content. It's just the website generated into the "out" folder. Should I just submit a PR?

**Comment by jameshughes89** [comment]
> > > > > > > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > > > > > > 
> > > > > > > > 
> > > > > > > > You can (a) build it locally or download the artifact from the action.
> > > > > > > 
> > > > > > > 
> > > > > > > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> > > > > > 
> > > > > > 
> > > > > > Just open the `index.html` file and navigate the website there. It will be a local version.
> > > > > 
> > > > > 
> > > > > Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.
> > > > 
> > > > 
> > > > First, you will need to add your lecture to the index in your PR Second, did you extract the website before opening the index? If not, that could be the issue.
> > > 
> > > 
> > > How do add my lecture to the PR index? And yes you're right, the error message was because of not extracting the files. Thanks!
> > 
> > 
> > Just look at the index currently and see how the others did it.
> 
> I have build and tested the website on my system, and it's working well with my lecture included in it. But I don't have see any changes to push because I haven't added any new content. It's just the website generated into the "out" folder. Should I just submit a PR?

You still need to add your content to the index page.

**Comment by AzizAbdi** [comment]
> > > > > > > > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > You can (a) build it locally or download the artifact from the action.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> > > > > > > 
> > > > > > > 
> > > > > > > Just open the `index.html` file and navigate the website there. It will be a local version.
> > > > > > 
> > > > > > 
> > > > > > Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.
> > > > > 
> > > > > 
> > > > > First, you will need to add your lecture to the index in your PR Second, did you extract the website before opening the index? If not, that could be the issue.
> > > > 
> > > > 
> > > > How do add my lecture to the PR index? And yes you're right, the error message was because of not extracting the files. Thanks!
> > > 
> > > 
> > > Just look at the index currently and see how the others did it.
> > 
> > 
> > I have build and tested the website on my system, and it's working well with my lecture included in it. But I don't have see any changes to push because I haven't added any new content. It's just the website generated into the "out" folder. Should I just submit a PR?
> 
> You still need to add your content to the index page.

Ok, I have added it to the index page. Now it's loading under student lectures section on my machine. What do I do next?

**Comment by jameshughes89** [comment]
> > > > > > > > > > > When should I do the "Test Sphinx Build Deploy"? It says skipped, I don't remember skipping. Is it after making the pull request or can I do it now for example? Because I am ready to push the changes requested from the earlier PR. Thanks!
> > > > > > > > > > 
> > > > > > > > > > 
> > > > > > > > > > You can (a) build it locally or download the artifact from the action.
> > > > > > > > > 
> > > > > > > > > 
> > > > > > > > > I have downloaded website from the action tab, but am not sure how to proceed from there. Is there a specific file I need to run to deploy?
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Just open the `index.html` file and navigate the website there. It will be a local version.
> > > > > > > 
> > > > > > > 
> > > > > > > Not sure I fully understand what I need to do. The `index.html` shows the website outline and when I click on any of them I get "Your file couldn’t be accessed" message. My lecture is in the student-lectures folder, but not on the local website, which I guess is what I need to change.
> > > > > > 
> > > > > > 
> > > > > > First, you will need to add your lecture to the index in your PR Second, did you extract the website before opening the index? If not, that could be the issue.
> > > > > 
> > > > > 
> > > > > How do add my lecture to the PR index? And yes you're right, the error message was because of not extracting the files. Thanks!
> > > > 
> > > > 
> > > > Just look at the index currently and see how the others did it.
> > > 
> > > 
> > > I have build and tested the website on my system, and it's working well with my lecture included in it. But I don't have see any changes to push because I haven't added any new content. It's just the website generated into the "out" folder. Should I just submit a PR?
> > 
> > 
> > You still need to add your content to the index page.
> 
> Ok, I have added it to the index page. Now it's loading under student lectures section on my machine. What do I do next?

Push the changes so you actually add it to the index in this PR.

**Comment by AzizAbdi** [comment]
I have pushed the changes to the `index.html` file. Is that what I needed to do?

**Comment by jameshughes89** [comment]
> I have pushed the changes to the `index.html` file. Is that what I needed to do?

No no --- remove the HTML file. You want the index.rst file u der the site directory.

**Comment by AzizAbdi** [comment]
I have reverted the changes to the `index.html` and modified the `index.rst`

**Review by jameshughes89 — APPROVED** [review]

**Comment by AzizAbdi** [comment]
It's still saying deploy skipped. Is there something else I should be doing to resolve this?

**Pull request merged**